ui/app/components/ui/mascot/mascot.component.js

```diff
@@ -46,6 +69,30 @@ export default class Mascot extends Component {
 
   componentDidMount () {
     this.mascotContainer.current.appendChild(this.logo.container)
+    this.directionTargetMap = directionTargetGenerator(this.mascotContainer.current.getBoundingClientRect())
+
+    const { lookAtTarget, lookAtDirection } = this.props
+
+    if (lookAtTarget?.x && lookAtTarget?.y) {
+      this.logo.lookAtAndRender(lookAtTarget)
+    } else if (lookAtDirection) {
+      this.logo.lookAtAndRender(this.directionTargetMap[lookAtDirection])
+    }
+  }
+
+  componentDidUpdate (prevProps) {
+    const { lookAtTarget: prevTarget = {}, lookAtDirection: prevDirection = '', followMouse: prevFollowMouse } = prevProps
+    const { lookAtTarget = {}, followMouse, lookAtDirection } = this.props
+
+    if (lookAtDirection && prevDirection !== lookAtDirection) {
+      this.logo.lookAtAndRender(this.directionTargetMap[lookAtDirection])
+    } else if (lookAtTarget?.x !== prevTarget?.x || lookAtTarget?.y !== prevTarget?.y) {
+      this.logo.lookAtAndRender(lookAtTarget)
+    }
+    if (followMouse && (prevFollowMouse !== followMouse)) {
```

Comment:
It seems that this would allow turning `followMouse` _on_, but not turning it off. Is this what was intended?

```suggestion
    if (prevFollowMouse !== followMouse) {
```